agent/command/s3_get.go

```diff
@@ -183,8 +183,25 @@ func (c *s3get) Execute(ctx context.Context,
 		attribute.Bool(s3GetTemporaryCredentialsAttribute, c.AwsSessionToken != ""),
 		attribute.String(s3GetRemoteFileAttribute, c.remoteFile),
 		attribute.String(s3GetExpandedRemoteFileAttribute, c.RemoteFile),
+		attribute.String(s3GetRoleARN, c.TemporaryRoleARN),
+		attribute.Bool(s3GetInternalBucketAttribute, utility.StringSliceContains(conf.InternalBuckets, c.Bucket)),
 	)
 
+	if c.TemporaryRoleARN != "" {
+		creds, err := comm.AssumeRole(ctx, c.taskdata, apimodels.AssumeRoleRequest{
+			RoleARN: c.TemporaryRoleARN,
+		})
+		if err != nil {
+			return errors.Wrap(err, "getting credentials for provided role arn")
```

Comment:
nit: May help debugging to include the name of the role ARN in the error message.